server.go

```diff
@@ -1369,6 +1369,13 @@ type Point struct {
 // WriteSeries writes series data to the database.
 // Returns the messaging index the data was written to.
 func (s *Server) WriteSeries(database, retentionPolicy string, points []Point) (uint64, error) {
+	// Make sure every point has at least one field.
+	for _, p := range points {
+		if len(p.Fields) == 0 {
+			return 0, fmt.Errorf("point %s has no fields", p.Name)
```

Comment:
OK, I see what you mean. Ironically `createMeasurementsIfNotExists` also loops through the points and I wrote that. 

Now that I see it, this function does a lot of stuff with a batch -- create shards, create series etc. It now seems to me you're on the right path, and that some basic validation of the batch is quite important before anything else happens. It kills me that we loop through the batch so many times. Oh well, we can profile it and see how it is, if it comes to it.